pkg/api/universal/conversation_middleware.go

```
@@ -0,0 +1,115 @@
+func NewStreamingPIIScrubber(windowSize int) (*StreamingPIIScrubber, error) {
+	scrubber := &StreamingPIIScrubber{
+		buffer:     make([]byte, 0, windowSize*2),
+		windowSize: windowSize,
+	}
+
+	// Initialize the default PII scrubber
+	defaultScrubber, err := piiscrubber.NewDefaultScrubber()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create default PII scrubber: %w", err)
+	}
+	scrubber.scrubber = defaultScrubber
+
+	return scrubber, nil
+}
+
+// ProcessChunk processes a chunk of streaming content with PII scrubbing
+func (s *StreamingPIIScrubber) ProcessChunk(chunk []byte) []byte {
+	if len(chunk) == 0 {
+		return chunk
+	}
+
+	// Append to buffer
+	s.buffer = append(s.buffer, chunk...)
+
+	// Process what we can safely process (leaving window for PII detection)
+	safeLength := len(s.buffer) - s.windowSize
+	if safeLength <= 0 {
+		return nil // Need more data
+	}
+
+	// Extract safe portion
+	safeContent := s.buffer[:safeLength]
+
+	// Apply PII scrubbing to safe portion
+	scrubbed := s.scrubPII(safeContent)
+
+	// Keep remaining buffer for next iteration
+	s.buffer = s.buffer[safeLength:]
+
+	return scrubbed
+}
+
+// Flush processes and returns any remaining content in the buffer
+func (s *StreamingPIIScrubber) Flush() []byte {
+	if len(s.buffer) == 0 {
+		return nil
+	}
+
+	// Process remaining buffer content
+	scrubbed := s.scrubPII(s.buffer)
+	s.buffer = s.buffer[:0] // Clear buffer
+
+	return scrubbed
+}
+
+// scrubPII applies PII scrubbing patterns to the content
+func (s *StreamingPIIScrubber) scrubPII(content []byte) []byte {
+	if s.scrubber == nil {
+		return content
+	}
+
+	// Convert to string for processing
+	text := string(content)
+
+	// Use the default scrubber to process the text
+	scrubbed, err := s.scrubber.ScrubTexts([]string{text})
+	if err != nil {
+		// If scrubbing fails, return original content
+		// TODO: Add logging here
+		return content
+	}
```

Why are we swallowing the error and failing silently?